### PR TITLE
RavenDB-6835 Splitting the conflict resolution work so the tx merger …

### PR DESCRIPTION
**Pull request by arekpalinski**
…just puts the resolutions (the conflicts are resolved earlier). Simplifying the code responsible for dealing with conflicts in the background.

### CODE REVIEW

File: src/Raven.Server/Documents/Replication/ResolveConflictOnReplicationConfigurationChange.cs [hunk 1]
```diff
@@ -105,56 +128,29 @@ public void RunConflictResolversOnce()
...
-        private class ResolveConflictsCommand : TransactionOperationsMerger.MergedTransactionCommand
+        
+        private class PutResolvedConflictsCommand : TransactionOperationsMerger.MergedTransactionCommand
         {
-            private readonly List<DocumentConflict> _conflicts;
+            private readonly List<DocumentConflict> _resolvedConflicts;
             private readonly ResolveConflictOnReplicationConfigurationChange _resolver;
 
-            public ResolveConflictsCommand(List<DocumentConflict> conflicts, ResolveConflictOnReplicationConfigurationChange resolver)
+            public PutResolvedConflictsCommand(List<DocumentConflict> resolvedConflicts, ResolveConflictOnReplicationConfigurationChange resolver)
```
Comment (ayende) [inline src/Raven.Server/Documents/Replication/ResolveConflictOnReplicationConfigurationChange.cs:137]: Where is the concurrency handling here?

Comment (arekpalinski) [reply src/Raven.Server/Documents/Replication/ResolveConflictOnReplicationConfigurationChange.cs:137]: What kind of concurrency errors should we expect here? Underneath, `PutResolvedDocument` just do the PUT with `null` as the expected etag? I guess it shouldn't?

Comment (ayende) [reply src/Raven.Server/Documents/Replication/ResolveConflictOnReplicationConfigurationChange.cs:137]: It shouldn't. Consider the case that you read the conflict at T1, then write it in T3.
But in T2, it was resolved externally.

Comment (arekpalinski) [reply src/Raven.Server/Documents/Replication/ResolveConflictOnReplicationConfigurationChange.cs:137]: So the handling will be just skipping the put if we got the concurrency exception, right?

**Pull request merged**